src/main/java/com/example/manager/controller/UserController.java

```diff
@@ -0,0 +1,143 @@
+package com.example.manager.controller;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.example.manager.repository.UserRepository;
+
+import com.example.manager.domain.User;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Service
+public class UserController {
+
+    public static final int maxFailedAttempts = 3;
+
+    // 60 Seconds
+    private static final long lockTimeDuration = 60 * 1000;
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Autowired
+    private UserRepository repo;
+
+    final private PasswordEncoder passwordEncoder = passwordEncoder();
+
+    public UserController() {
+        super();
+    }
+
+
+    public User register(User user) {
+        return repo.save(user);
+    }
+
+    public boolean checkIfUsernameTaken(String username) {
+        return repo.existsById(username);
+    }
+
+
+    public boolean login(String username, String password) {
+        User temp = null;
+        try {
+            temp = repo.findById(username).get();

```

Comment:
findById is returning an Optional. 

`repo.findById(username).orElseThrow()` would be more elegant.